core/src/main/java/com/graphhopper/util/exceptions/ConnectionNotFoundException.java

```diff
@@ -25,7 +30,75 @@
  * @author Peter Karich
  */
 public class ConnectionNotFoundException extends DetailedIllegalArgumentException {
+    private static final String INTRO = "Connection between ";
+    private static final String FMT = "point %d and point %d";
+    private static final String JOIN = ", and ";
+    private static final String OUTRO = " not found";
+
     public ConnectionNotFoundException(String var1, Map<String, Object> details) {
         super(var1, details);
     }
+
+    public ConnectionNotFoundException(List<Path> paths, PointList waypoints) {
+        super(unconnectedListString(paths), unconnectedInfo(paths, waypoints));
+    }
+
+    private static String unconnectedListString(List<Path> paths) {
+        StringBuilder sb = new StringBuilder(INTRO);
+
+        for (int n = 0; n < paths.size(); ++n) {
+            if (!paths.get(n).isFound())
+                sb.append(String.format(FMT, n, n + 1)).append(JOIN);
+        }
+
+        if (sb.length() == INTRO.length())
+            throw new IllegalArgumentException("All provided paths are found");
+
+        sb.setLength(sb.length() - JOIN.length()); // delete last occurrence of JOIN
+        sb.append(OUTRO);
+
+        return sb.toString();
+    }
+
+    private static Map<String, Object> unconnectedInfo(List<Path> paths, PointList waypoints) {
+        Map<String, Object> info = new HashMap<>();
+
+        for (int pathIndex = 0; pathIndex < paths.size(); ++pathIndex) {
+            if (!paths.get(pathIndex).isFound()) {
+                Map<String, Number> infoEntry = new HashMap<>();
+
+                infoEntry.put(Keys.idFrom.name(), pathIndex);
+                infoEntry.put(Keys.latFrom.name(), waypoints.getLat(pathIndex));
+                infoEntry.put(Keys.lonFrom.name(), waypoints.getLon(pathIndex));
+                infoEntry.put(Keys.idTo.name(), pathIndex + 1);
+                infoEntry.put(Keys.latTo.name(), waypoints.getLat(pathIndex + 1));
+                infoEntry.put(Keys.lonTo.name(), waypoints.getLon(pathIndex + 1));
+
+                info.put("connection-not-found-" + info.size(), infoEntry);
```

Comment:
I know this was my own idea :), but maybe it would be better to create json like:
```
connection-not-found: [
{...},
{...}
]
```

WDYT @karussell?

Reply:
No sweat, API design needs to be thought through :-)

When I'm back from vacation and you guys have decided on your format of choice I can adjust it.

Reply:
Sorry for the delay will have a look in the next time!